src/acrcssc/azext_acrcssc/helper/_taskoperations.py

```diff
@@ -0,0 +1,545 @@
+    ERROR_MESSAGE_WORKFLOW_TASKS_DOES_NOT_EXIST,
+    ERROR_MESSAGE_WORKFLOW_TASKS_ALREADY_EXISTS,
+    CSSC_WORKFLOW_POLICY_REPOSITORY,
+    CONTINUOUSPATCH_TASK_PATCHIMAGE_NAME,
+    CONTINUOUSPATCH_TASK_SCANIMAGE_NAME,
+    DESCRIPTION,
+    WORKFLOW_VALIDATION_MESSAGE,
+    TaskRunStatus)
+from azure.cli.core.azclierror import AzCLIError, InvalidArgumentValueError, ResourceNotFoundError
+from azure.cli.core.commands import LongRunningOperation
+from azure.cli.core.commands.progress import IndeterminateProgressBar
+from azure.cli.command_modules.acr._utils import prepare_source_location
+from azure.cli.command_modules.acr._archive_utils import logger as acr_archive_utils_logger
+from azure.core.exceptions import HttpResponseError
+from azure.mgmt.core.tools import parse_resource_id
+from azext_acrcssc._client_factory import cf_acr_tasks, cf_authorization, cf_acr_registries_tasks, cf_acr_runs
+from azext_acrcssc.helper._deployment import validate_and_deploy_template
+from azext_acrcssc._validators import check_continuous_task_exists, check_continuous_task_config_exists
+from datetime import datetime, timezone, timedelta
+from ._utility import convert_timespan_to_cron, convert_cron_to_schedule, create_temporary_dry_run_file, delete_temporary_dry_run_file
+from azext_acrcssc.helper._ociartifactoperations import create_oci_artifact_continuous_patch, get_oci_artifact_continuous_patch, delete_oci_artifact_continuous_patch
+from ._workflow_status import WorkflowTaskStatus
+
+logger = get_logger(__name__)
+
+
+def create_update_continuous_patch_v1(cmd,
+                                      registry,
+                                      cssc_config_file,
+                                      schedule,
+                                      dryrun,
+                                      run_immediately,
+                                      is_create_workflow=True):
+
+    logger.debug(f"Entering continuousPatchV1_creation {cssc_config_file} {dryrun} {run_immediately}")
+
+    resource_group = parse_resource_id(registry.id)[RESOURCE_GROUP]
+    schedule_cron_expression = None
+    cssc_tasks_exists, task_list = check_continuous_task_exists(cmd, registry)
+
+    if schedule is not None:
+        schedule_cron_expression = convert_timespan_to_cron(schedule)
+        logger.debug(f"converted schedule to cron expression: {schedule_cron_expression}")
+
+    if is_create_workflow:
+        if cssc_tasks_exists:
+            raise AzCLIError(f"{ERROR_MESSAGE_WORKFLOW_TASKS_ALREADY_EXISTS}")
+        _create_cssc_workflow(cmd, registry, schedule_cron_expression, resource_group, dryrun)
+    else:
+        if not cssc_tasks_exists:
+            raise AzCLIError(f"{ERROR_MESSAGE_WORKFLOW_TASKS_DOES_NOT_EXIST}")
+
+        _update_cssc_workflow(cmd, registry, schedule_cron_expression, resource_group, dryrun, task_list)
+
+    if cssc_config_file is not None:
+        create_oci_artifact_continuous_patch(registry, cssc_config_file, dryrun)
```

Comment:
Per Sajay’s comments, the OCI artifact should be pushed before the task is created. If this step fails, the process should simply exit.

Reply:
fixed on https://github.com/AzureCR/azure-cli-extensions/pull/45